### PR TITLE
Implement AmountService

### PR DESCRIPTION
**Pull request by antooni**
TODO:
- [x] tests
- [ ] resolves TODOs

Resolves L2B-4716

### CODE REVIEW

File: packages/backend/src/modules/tvl2/AmountService.ts [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+
+export interface AmountServiceDependencies {
+  readonly rpcClient: RpcClient
+  readonly multicallClient: MulticallClient
+  readonly erc20Codec: ERC20MulticallCodec
+  readonly nativeAssetCodec?: NativeAssetMulticallCodec
+  logger: Logger
+}
+
+interface Amount {
```
Comment (antooni) [inline packages/backend/src/modules/tvl2/AmountService.ts:33]: is this type needed?

File: packages/backend/src/modules/tvl2/AmountService.ts [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+    blockNumber: number,
+    timestamp: UnixTime,
+  ): Promise<AmountRecord[]> {
+    const nativeAssetCodecAtBlock = getNativeAssetCodecAtBlock(
+      this.$.nativeAssetCodec,
+      blockNumber,
+    )
+    const [forRpc, forMulticall] = partition(
+      configurations,
+      // isNotSupportedByMulticall checks that type is "escrow"
```
Comment (antooni) [inline packages/backend/src/modules/tvl2/AmountService.ts:54]: this comment is probably not needed any more

File: packages/shared-pure/src/types/AmountConfigEntry.ts [hunk 3]
```diff
@@ -25,7 +25,7 @@ export interface EscrowEntry extends AmountConfigBase {
   escrowAddress: EthereumAddress
 }
 
-interface AmountConfigBase {
+export interface AmountConfigBase {
```
[the remark this answers is not in the document]
Comment (antooni) [reply packages/shared-pure/src/types/AmountConfigEntry.ts:28]: nvm, I see now

File: packages/backend/src/modules/tvl2/AmountService.ts (hunk 2 above)
Comment (michalsidzej) [reply packages/backend/src/modules/tvl2/AmountService.ts]: [91152b1](https://github.com/l2beat/l2beat/pull/3266/commits/91152b1b98d4c042d116ffc77cbc49031e2ed700)

File: packages/backend/src/modules/tvl2/AmountService.ts (hunk 1 above)
Comment (michalsidzej) [reply packages/backend/src/modules/tvl2/AmountService.ts:33]: To define return types. Helps a lot when pinpointing bug in TS.

**Review by michalsidzej — APPROVED** [review]

**Pull request merged**